src/content/docs/en/guides/cms/keystatic.mdx

```diff
@@ -2,14 +2,328 @@
+</Aside>
+
+You *could* do this, but instead, use Astro's new [`hybrid` rendering mode](/en/guides/server-side-rendering/#converting-a-static-site-to-hybrid-rendering), which works even better here.
+
+### Astro's SSR hybrid mode
+
+In the `astro.config.mjs`, add the `output: 'hybrid'` option:
+
+```diff title="astro.config.mjs"
+export default defineConfig({
++ output: 'hybrid',
+  integrations: [react()],
+})
+```
+
+Hybrid rendering will still try to pre-render every page by default, with a mechanism to opt out of it.
+
+You *don't want* prerendering on Keystatic routes. Export a `const prerender = false` in your Keystatic pages file:
+
+```diff
+---
+// src/pages/keystatic/[...params].astro
+import { Keystatic } from '../../../keystatic.page'
+
++ export const prerender = false
+---
+
+<Keystatic client:only />
+
+```
+
+Before you can read and write data, you also need to create some `API routes` for Keystatic
+
+### Keystatic API Routes
+
+Create a new file called `src/pages/api/keystatic/[...params].ts`
+
+Once again, use Astro's Rest parameters here:
+
+```jsx
+// src/pages/api/keystatic/[...params].ts
+import { makeHandler } from '@keystatic/astro/api'
+import keystaticConfig from '../../../../keystatic.config'
+
+export const all = makeHandler({
+  config: keystaticConfig,
+})
+
+export const prerender = false
+```
+
+Notice those routes also opt out of prerendering.
+
+Aaaand... You should be all set to use Keystatic now 🎉
+
+:::note
+At this point, you should be able to visit the `/keystatic` route in the browser and see the Keystatic Admin UI. 
+:::
+
+## Creating a new post
+
+Try and visit the `/keystatic` page in the browser one more time, and click on the “Posts” collection.
+
+Go ahead, and create a new post.
+
+:::note
+In our Keystatic config file, you've set the `path` property for our `posts` collection to `src/content/posts/*`.
+
+As a result, creating a new post from the Keystatic Admin UI should create a new `content` directory in the `src` directory, with the new post `.mdoc` file inside.
+:::
+
+If everything worked correctly, you'll find your new post inside the `src/content/posts` directory:
+
+<FileTree title="Project Structure">
+- src/
+  - content/
+    - posts/
+      - **my-first-post.mdoc**
+</FileTree>
+
+That Markdoc file should look something like this:
+
+```markdown
+---
+title: My First Post
+---
+
+This is my very first post. I am **super** excited.
+```
+
+Niiiice ✨
+
+Time to display that post on the frontend now.
+
+## Rendering Keystatic content
+
+:::tip
+While Keystatic provides its own [Reader and Renderer APIs](https://keystatic.com/docs/reader-api) to pull data your frontend, it also work great with Astro's [content collections](/en/guides/content-collections/)!
+:::
+
+### Displaying a collection list
+
+Since you told Keystatic to manage `posts` in `src/content/posts`, they'll be available as Astro's content collections as the `posts` collection slug:
```

Comment:
```suggestion
Since you instructed Keystatic to manage `posts` in `src/content/posts`, they'll be available as Astro's content collections as the `posts` collection slug:
```